agent/src/main/java/kanela/agent/InstrumentationLoader.java

```diff
@@ -51,10 +55,14 @@
         });
     }
 
-    private static Option<KanelaInstrumentation> loadInstrumentation(String instrumentationClassName, ClassLoader classLoader) {
+    private static Option<KanelaInstrumentation> loadInstrumentation(KanelaConfiguration configuration, KanelaConfiguration.ModuleConfiguration moduleConfiguration, String instrumentationClassName, ClassLoader classLoader) {
         Logger.info(() -> format(" ==> Loading {0} ", instrumentationClassName));
         try {
-            return Option.some((KanelaInstrumentation) Class.forName(instrumentationClassName, true, classLoader).newInstance());
+            Class<?> instrumentationClass = Class.forName(instrumentationClassName, true, classLoader);
```

Comment:
I think that the responsibility of register the information of the intrumentations should not be here, it seems to me better to move this code inside `KanelaInstrumentation` and leave this code clean,  wdyt guys?

Reply:
This is the only place when we have the loaded class of the instrumentation together with the moduleConfiguration and the kanelaConfig. May be this can be on another function called on the bigger maps/flatmaps/filter in the `load` function.